### PR TITLE
assessment first attempt

### PR DESCRIPTION
**Pull request by henrike32**
# Submitting Week 5 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [x] JavaScript Coding Challenges
- [x] Ruby Coding Challenges
- [x] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:
Didn't manage to finish the JavaScript challenges.

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -13,40 +13,134 @@
...
+        expect(gotString(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect(gotString(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
+
+// ReferenceError: gotString is not defined
+// b) Create the function that makes the test pass.
+// pseudocode - 
+// output -> takes in a string
+// input -> returns a string with a coded message which converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.
```
Comment (chrisyhlee) [inline code-challenges.test.js:34]: input and output are reversed

File: code-challenges.test.js [hunk 2]
```diff
@@ -13,40 +13,134 @@
...
+// process -
+// create a function named 'gotString'
+// using the .replace method we can replace the given letters/integers in the original string ->  str1.replace(regexp, str2)
+// the callback function (match) => codeMap[match] is used to change each matched letter to its corresponding code from the codeMap object
+// Method 1:
```
Comment (chrisyhlee) [inline code-challenges.test.js:39]: Based on your pseudo code, it seems like you don't fully understand what is happening with this solution.
- make an object with key value pairs representing letters to match and their replacement
- use .replace to iterate through string
- use regex with the letters to look for, using the g flag to find all occurrences, not just the first
- callback function will find each match and replace the matched letter with the value in the object

File: code-challenges.test.js [hunk 3]
```diff
@@ -13,40 +13,134 @@
...
+
+// create a function named 'filterArray'
+const filterArray = (arr, letter) => {
+// create a variable to store the new array with the expected letters
+let newArray = [];
+
+    // use a for loop to iterate through the array
+    for (let i=0; i < arr.length; i++){
+        // locate the letter we want using .includes
+        if (arr[i].includes(letter)){
```
Comment (chrisyhlee) [inline code-challenges.test.js]: Great thought process, but what about the capital A in 'Apricot'? This currently won't push that word into the newArray.

File: ruby-challenges.rb [hunk 4]
```diff
@@ -2,29 +2,72 @@
...
-filter_letter_o = 'o'
+# beverages_array = ['coffee', 'tea', 'juice', 'water', 'soda water']
+# filter_letter_o = 'o'
 # Expected output: ['coffee', 'soda water']
-filter_letter_t = 't'
+# filter_letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
+# I googled  a method to filter an array and discovered the .select method, I used the .select method and the .include method to get the stringsI wanted inside the 'beverages_array'
 
+# result1 = beverages_array.select! { |string| string.include?(filter_letter_o) }
```
Comment (chrisyhlee) [inline ruby-challenges.rb:27]: Keep in mind that the prompt is asking you to create a method, not a variable

File: ruby-challenges.rb [hunk 5]
```diff
@@ -2,29 +2,72 @@
...
+        @current_speed = 0
+
+    end
+
+    def bike_info
+        "The #{:model} has #{:wheels} and is going #{:current_speed} mph."
+    end
+    # -------------------3b) Add the ability to pedal faster and brake. The pedal_faster method should increase the speed by a given amount. The brake method should decrease the speed by a given amount. The bike cannot go negative speeds.
+    def pedal_faster(current_speed)
+        # had to find a way to increase the initial speed and maintain the value, didn't knew if I could use 'Integer' as a incremental value, tried this out and it worked! Happy to actually do some code without referring to chatGPT or google.
```
Comment (chrisyhlee) [inline ruby-challenges.rb:75]: Big win!! Nice work!

File: ruby-challenges.rb [hunk 6]
```diff
@@ -2,29 +2,72 @@
...
+        @current_speed =  @current_speed + current_speed
+    end 
 
+    def pedal_brake(break_speed)
+        # in this scenario you want the speed to reduce but without going to zero, so I did a conditional statement to stop the operation when the speed would go to less than zero.
+        @current_speed = @current_speed - break_speed 
+# Didn't figured it out why I am getting ' 10 ,28 , nil , "negative speed doesn't exist" ' as an output instead of getting 23 I am getting nil, without lines of code 57-59 the code gives me the correct expected values except for the -2 one. Since I still have a lot to cover in this assessment I will keep pressing forward.
+        # if @current_speed < 0
+        #     "Negative speed doesn't exist"
```
Comment (chrisyhlee) [inline ruby-challenges.rb:84]: Remember you still want to set current_speed to something. The lowest it can be is 0

**Review by chrisyhlee — CHANGES_REQUESTED** [review]
Waiting for code challenge 3

**Review by chrisyhlee — COMMENTED** [review]
I see you added your classmates' approaches to your own. This is great, but only if you are taking the time to understand each approach and learning from them. If you don't, you are not gaining anything out of it, which is only hurting yourself in the end. I am available to reach out to at any time, and I highly suggest you go to your mentor with these approaches and any concepts you don't understand! This displays Novice level currently. I believe in you Henri, keep working hard!

**Comment by henrike32** [comment]
I've been paying attention to their approach on the coding part, most of the codes I've added to the assessment refactor the owner explained during the assessment review. Later on I will make a single file with all of these ways to do certain functions , so I can always refer when needed. Also thank you for the support!! I really appreciate you for supporting me here!

**Pull request merged**